pulsar-common/src/main/java/org/apache/pulsar/common/naming/TopicName.java

```diff
@@ -109,11 +109,15 @@ private TopicName(String completeTopicName) {
                 // The short topic name can be:
                 // - <topic>
                 // - <property>/<namespace>/<topic>
+                // When the topic is transaction topic, the topic name is:
+                // - <property>/<namespace>/<topic>/_txnlog
                 String[] parts = StringUtils.split(completeTopicName, '/');
                 if (parts.length == 3) {
                     completeTopicName = TopicDomain.persistent.name() + "://" + completeTopicName;
                 } else if (parts.length == 1) {
                     completeTopicName = TopicDomain.persistent.name() + "://" + PUBLIC_TENANT + "/" + DEFAULT_NAMESPACE + "/" + parts[0];
+                } else if (parts.length == 4 && completeTopicName.endsWith("/_txnlog")) {
```

Comment:
I still not convinced why do you need this change here. This is a public api to every Pulsar users. How we store `_txnlog` should be done within the broker. The user should not be aware of this. Can't we just deal with the naming in broker itself?